controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -681,3 +935,51 @@ func diffAddresses(oldAddresses, newAddresses AddressSet) (add, remove AddressSe
 	}
 	return
 }
+
+func isTargetPortInSlice(targetPort namedPort, slicePorts []discovery.EndpointPort) bool {
+	for _, p := range slicePorts {
+		switch targetPort.Type {
+		case intstr.Int:
+			if *p.Port == targetPort.IntVal {
```

Comment:
What happens if this is nil? The docs say something intersting that I am not sure I can interpret correctly: 

```
	// ports specifies the list of network ports exposed by each endpoint in
	// this slice. Each port must have a unique name. When ports is empty, it
	// indicates that there are no defined ports. When a port is defined with a
	// nil port value, it indicates "all ports". Each slice may include a
	// maximum of 100 ports.
	// +optional
	// +listType=atomic
```

All ports?... not sure what they mean here. Any idea?

Reply:
@zaharidichev I looked into this and I couldn't find anything that makes sense. I'm a bit puzzled by it too, in all honesty.

I had to do some work around the nil port today, it's not related to the description but it might offer some insight. In my case, whenever an endpointslice had all of its associated pods, all of the addresses (i.e endpoints) and ports automatically got deleted and set to nil (because they use pointers instead of empty lists *sigh*).

I suppose in this case, an EndpointSlice with a nil `ports` field, targets all targetPorts of the service. So, going back on a previous explanation, an `EndpointSlice` is an endpoints subset. An endpoints subset is a grouping of addresses with  common ports. This means every slice (assuming none of the slices hit their endpoint capacity limit) would have a different set of ports, an EndpointSlice with nil ports then would mean ALL ports of the service? Perhaps for the rare case when a pod exposes all ports? Still, not sure why it has to be nil, but it might make sense? Lemme know what you think.

Reply:
>The port number of the endpoint. If this is not specified, ports are not restricted and must be interpreted in the context of the specific consumer.

It sounds a bit underspecified.  Let's not worry too much about this.  We can come back and adjust this behavior if we run into a situation where it's problematic.